cmd/serve.go

```diff
@@ -90,31 +107,70 @@ func serve(ctx context.Context) {
 	}
 
 	client := ent.NewClient(cOpts...)
+	defer client.Close()
 
+	var middleware []echo.MiddlewareFunc
+
+	// jwt auth middleware
 	if viper.GetBool("oidc.enabled") {
 		auth, err := echojwtx.NewAuth(ctx, config.AppConfig.OIDC)
 		if err != nil {
 			logger.Fatal("failed to initialize jwt authentication", zap.Error(err))
 		}
 
-		auth.JWTConfig.Skipper = echox.SkipDefaultEndpoints
-		config.AppConfig.Server = config.AppConfig.Server.WithMiddleware(auth.Middleware())
+		middleware = append(middleware, auth.Middleware())
 	}
 
-	srv, err := echox.NewServer(logger.Desugar(), config.AppConfig.Server, versionx.BuildDetails())
+	srv, err := echox.NewServer(logger.Desugar(), echox.ConfigFromViper(viper.GetViper()), versionx.BuildDetails())
```

Comment:
Aren't we heading away from the `ConfigFromViper` helpers. I think we just want everything loaded from the `Config` object?